### ISSUE TITLE
Message type "inference.ModelParameter" has no field named "Ultralytics" when using yoloe with triton inference server

### ISSUE THREAD

**Issue by arbitropy** [ultralytics/ultralytics#22111]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.


### Ultralytics YOLO Component

Export, Integrations

### Bug

I am trying to run triton inference server using the official documentation, and the doc works perfectly well for yolo11n, but i face the following error while doing the same for yoloe:
```
$ docker run --runtime=nvidia --gpus all -v ./tmp/triton_repo:/models -p 8000:8000 nvcr.io/nvidia/tritonserver:24.09-py3 tritonserver --model-repository=/models

=============================
== Triton Inference Server ==
=============================

NVIDIA Release 24.09 (build 112408254)
Triton Server Version 2.50.0

Copyright (c) 2018-2024, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.

Various files include modifications (c) NVIDIA CORPORATION & AFFILIATES.  All rights reserved.

This container image and its contents are governed by the NVIDIA Deep Learning Container License.
By pulling and using the container, you accept the terms and conditions of this license:
https://developer.nvidia.com/ngc/nvidia-deep-learning-container-license

I0916 23:21:47.864641 1 pinned_memory_manager.cc:277] "Pinned memory pool is created at '0x7517e0000000' with size 268435456"
I0916 23:21:47.866874 1 cuda_memory_manager.cc:107] "CUDA memory pool is created on device 0 with size 67108864"
[libprotobuf ERROR /tmp/tritonbuild/tritonserver/build/_deps/repo-third-party-build/grpc-repo/src/grpc/third_party/protobuf/src/google/protobuf/text_format.cc:337] Error parsing text-format inference.ModelConfig: 6:49: Message type "inference.ModelParameter" has no field named "Ultralytics".
E0916 23:21:47.870936 1 model_repository_manager.cc:1460] "Poll failed for model directory 'yoloe-11l-seg': failed to read text proto from /models/yoloe-11l-seg/config.pbtxt"
I0916 23:21:47.870959 1 model_lifecycle.cc:472] "loading: yolo11n:1"
I0916 23:21:47.872204 1 onnxruntime.cc:2875] "TRITONBACKEND_Initialize: onnxruntime"
I0916 23:21:47.872212 1 onnxruntime.cc:2885] "Triton TRITONBACKEND API version: 1.19"
I0916 23:21:47.872215 1 onnxruntime.cc:2891] "'onnxruntime' TRITONBACKEND API version: 1.19"
I0916 23:21:47.872217 1 onnxruntime.cc:2921] "backend configuration:\n{\"cmdline\":{\"auto-complete-config\":\"true\",\"backend-directory\":\"/opt/tritonserver/backends\",\"min-compute-capability\":\"6.000000\",\"default-max-batch-size\":\"4\"}}"
I0916 23:21:47.881368 1 onnxruntime.cc:2986] "TRITONBACKEND_ModelInitialize: yolo11n (version 1)"
I0916 23:21:49.267597 1 onnxruntime.cc:3051] "TRITONBACKEND_ModelInstanceInitialize: yolo11n_0 (GPU device 0)"
I0916 23:21:50.345311 1 model_lifecycle.cc:839] "successfully loaded 'yolo11n'"
I0916 23:21:50.345343 1 server.cc:604] 
+------------------+------+
| Repository Agent | Path |
+------------------+------+
+------------------+------+

I0916 23:21:50.345357 1 server.cc:631] 
+-------------+-----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Backend     | Path                                                            | Config                                                                                                                                                        |
+-------------+-----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+
| onnxruntime | /opt/tritonserver/backends/onnxruntime/libtriton_onnxruntime.so | {"cmdline":{"auto-complete-config":"true","backend-directory":"/opt/tritonserver/backends","min-compute-capability":"6.000000","default-max-batch-size":"4"}} |
+-------------+-----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+

I0916 23:21:50.345371 1 server.cc:674] 
+---------+---------+--------+
| Model   | Version | Status |
+---------+---------+--------+
| yolo11n | 1       | READY  |
+---------+---------+--------+

I0916 23:21:50.382000 1 metrics.cc:877] "Collecting metrics for GPU 0: NVIDIA GeForce RTX 3090"
I0916 23:21:50.385173 1 metrics.cc:770] "Collecting CPU metrics"
I0916 23:21:50.385221 1 tritonserver.cc:2598] 
+----------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Option                           | Value                                                                                                                                                                                                           |
+----------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| server_id                        | triton                                                                                                                                                                                                          |
| server_version                   | 2.50.0                                                                                                                                                                                                          |
| server_extensions                | classification sequence model_repository model_repository(unload_dependents) schedule_policy model_configuration system_shared_memory cuda_shared_memory binary_tensor_data parameters statistics trace logging |
| model_repository_path[0]         | /models                                                                                                                                                                                                         |
| model_control_mode               | MODE_NONE                                                                                                                                                                                                       |
| strict_model_config              | 0                                                                                                                                                                                                               |
| model_config_name                |                                                                                                                                                                                                                 |
| rate_limit                       | OFF                                                                                                                                                                                                             |
| pinned_memory_pool_byte_size     | 268435456                                                                                                                                                                                                       |
| cuda_memory_pool_byte_size{0}    | 67108864                                                                                                                                                                                                        |
| min_supported_compute_capability | 6.0                                                                                                                                                                                                             |
| strict_readiness                 | 1                                                                                                                                                                                                               |
| exit_timeout                     | 30                                                                                                                                                                                                              |
| cache_enabled                    | 0                                                                                                                                                                                                               |
+----------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

I0916 23:21:50.385237 1 server.cc:305] "Waiting for in-flight requests to complete."
I0916 23:21:50.385242 1 server.cc:321] "Timeout 30: Found 0 model versions that have in-flight inferences"
I0916 23:21:50.385287 1 server.cc:336] "All models are stopped, unloading models"
I0916 23:21:50.385289 1 server.cc:345] "Timeout 30: Found 1 live models and 0 in-flight non-inference requests"
I0916 23:21:50.385559 1 onnxruntime.cc:3103] "TRITONBACKEND_ModelInstanceFinalize: delete instance state"
I0916 23:21:50.556097 1 onnxruntime.cc:3027] "TRITONBACKEND_ModelFinalize: delete model state"
I0916 23:21:50.556154 1 model_lifecycle.cc:624] "successfully unloaded 'yolo11n' version 1"
I0916 23:21:51.385400 1 server.cc:345] "Timeout 29: Found 0 live models and 0 in-flight non-inference requests"
error: creating server: Internal - failed to load all models
```
the generataed protobuf config is given below:
```

# Add metadata
parameters {
  key: "metadata"
  value {
    string_value: "{'description': "Ultralytics YOLOe-11l-seg model trained on {'train': {'yolo_data': ['Objects365v1.yaml'], 'grounding_data': [{'img_path': '../datasets/flickr/full_images/', 'json_file': '../datasets/flickr/annotations/final_flickr_separateGT_train_segm.json'}, {'img_path': '../datasets/mixed_grounding/gqa/images', 'json_file': '../datasets/mixed_grounding/annotations/final_mixed_train_no_coco_segm.json'}]}, 'val': {'yolo_data': ['lvis.yaml']}}", 'author': 'Ultralytics', 'date': '2025-09-17T05:21:42.314054', 'version': '8.3.200', 'license': 'AGPL-3.0 License (https://ultralytics.com/license)', 'docs': 'https://docs.ultralytics.com', 'stride': 32, 'task': 'segment', 'batch': 16, 'imgsz': [640, 640], 'names': {0: 'person', 1: 'bus'}, 'args': {'batch': 16, 'half': True, 'dynamic': True, 'simplify': True, 'opset': None, 'nms': False}, 'channels': 3}"
  }
}

# Enable TensorRT for GPU inference
# First run will be slow due to TensorRT engine conversion
optimization {
  execution_accelerators {
    gpu_execution_accelerator {
      name: "tensorrt"
      parameters {
        key: "precision_mode"
        value: "FP16"
      }
      parameters {
        key: "max_workspace_size_bytes"
        value: "3221225472"
      }
      parameters {
        key: "trt_engine_cache_enable"
        value: "1"
      }
      parameters {
        key: "trt_engine_cache_path"
        value: "/models/yoloe-11l-seg/1"
      }
    }
  }
}
```
Here is the export code:
```
from ultralytics import YOLO, YOLOE
yoloe_model = YOLOE("yoloe-11l-seg.pt", task="segment")
names = ["person", "bus"]
yoloe_model.set_classes(names, yoloe_model.get_text_pe(names))
metadata = []

def export_cb(exporter):
    metadata.append(exporter.metadata)

yoloe_file = yoloe_model.export(
    format="onnx",
    half=True,
    batch=16,
    dynamic=True,
    device="cuda:0"
)
# The rest are path operations
```
What is the cause of this? 
Can I manually configure the config file to mitigate this?

### Environment

```
Ultralytics 8.3.200 🚀 Python-3.12.11 torch-2.8.0+cu128 CUDA:0 (NVIDIA GeForce RTX 3090, 24135MiB)
Setup complete ✅ (32 CPUs, 62.6 GB RAM, 1707.2/1831.7 GB disk)

OS                     Linux-6.8.0-79-generic-x86_64-with-glibc2.35
Environment            Linux
Python                 3.12.11
Install                git
Path                  triton-ultralytics/.venv/lib/python3.12/site-packages/ultralytics
RAM                    62.56 GB
Disk                   1707.2/1831.7 GB
CPU                    Intel Core(TM) i9-14900K
CPU count              32
GPU                    NVIDIA GeForce RTX 3090, 24135MiB
GPU count              1
CUDA                   12.8

numpy                  ✅ 2.3.3>=1.23.0
matplotlib             ✅ 3.10.6>=3.3.0
opencv-python          ✅ 4.11.0.86>=4.6.0
pillow                 ✅ 11.3.0>=7.1.2
pyyaml                 ✅ 6.0.2>=5.3.1
requests               ✅ 2.32.5>=2.23.0
scipy                  ✅ 1.16.2>=1.4.1
torch                  ✅ 2.8.0>=1.8.0
torch                  ✅ 2.8.0!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.23.0>=0.9.0
psutil                 ✅ 7.0.0
polars                 ✅ 1.33.1
ultralytics-thop       ✅ 2.0.17>=2.0.0
```

### Minimal Reproducible Example

```
from ultralytics import YOLO, YOLOE
yoloe_model = YOLOE("yoloe-11l-seg.pt", task="segment")
names = ["person", "bus"]
yoloe_model.set_classes(names, yoloe_model.get_text_pe(names))
metadata = []

def export_cb(exporter):
    metadata.append(exporter.metadata)

yoloe_file = yoloe_model.export(
    format="onnx",
    half=True,
    batch=16,
    dynamic=True,
    device="cuda:0"
)

yolo_model_name = "yolo11n"
triton_repo_path = Path("tmp") / "triton_repo"
yoloe_model_name = "yoloe-11l-seg"
triton_yoloe_model_path = triton_repo_path / yoloe_model_name
(triton_yoloe_model_path / "1").mkdir(parents=True, exist_ok=True)
from pathlib import Path
Path(yoloe_file).rename(triton_yoloe_model_path / "1" / "model.onnx")
(triton_yoloe_model_path / "config.pbtxt").touch()

yoloe_data = """
# Add metadata
parameters {
  key: "metadata"
  value {
    string_value: "%s"
  }
}

# Enable TensorRT for GPU inference
# First run will be slow due to TensorRT engine conversion
optimization {
  execution_accelerators {
    gpu_execution_accelerator {
      name: "tensorrt"
      parameters {
        key: "precision_mode"
        value: "FP16"
      }
      parameters {
        key: "max_workspace_size_bytes"
        value: "3221225472"
      }
      parameters {
        key: "trt_engine_cache_enable"
        value: "1"
      }
      parameters {
        key: "trt_engine_cache_path"
        value: "/models/yoloe-11l-seg/1"
      }
    }
  }
}
""" % metadata[0]  # noqa 

with open(triton_yoloe_model_path / "config.pbtxt", "w") as f:
    f.write(yoloe_data)

import contextlib
import subprocess
import time
from tritonclient.http import InferenceServerClient

# Define image https://catalog.ngc.nvidia.com/orgs/nvidia/containers/tritonserver
tag = "nvcr.io/nvidia/tritonserver:24.09-py3"

# Pull the image
subprocess.call(f"docker pull {tag}", shell=True)

# Run the Triton server and capture the container ID
container_id = (
    subprocess.check_output(
        f"docker run -d --runtime=nvidia --gpus all -v {triton_repo_full_path}:/models -p 8000:8000 {tag} tritonserver --model-repository=/models",
        shell=True,
    )
    .decode("utf-8")
    .strip()
)
print(f"Triton server started with container ID: {container_id}")
print(f"To view logs, run: docker logs -f {container_id}")

# Wait for the Triton server to start
triton_client = InferenceServerClient(url="localhost:8000", verbose=False, ssl=False)

# Wait until model is ready
for _ in range(10):
    with contextlib.suppress(Exception):
        assert triton_client.is_model_ready(yoloe_model_name)
        break
    time.sleep(1)


### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @arbitropy, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get unblocked quickly—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for the detailed logs and information! If this is a 🐛 Bug Report, your shared snippet is helpful. If possible, please also provide or confirm a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that others can run end-to-end (a small repo or script showing export + Triton config + launch), keeping only the essentials.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
`set_classes` doesn't work after export.

**Comment by Y-T-G** [comment]
You can run `metadata[0].pop("description")` after your export before the config is created

**Comment by arbitropy** [comment]
> You can run `metadata[0].pop("description")` after your export before the config is created

This solves it. Is there a reason why the description has to be removed for yoloe but not for yolo11n? 

Also, is there a way to set timeout value in prediction? for each different batch size, there is a initial delay in inference, which i assume is because triton is optimizing the model for that specific batch size. but the predicts always get timed out due to it for the first few times.

**Comment by Y-T-G** [comment]
Probably the braces in the description causing parsing issue.
